test/optimizations/constDomain/basic.chpl

```diff
@@ -0,0 +1,94 @@
+use BlockDist;
+
+proc test(d) {
+  var arr: [d] int;
+
+  writeln("Using user-variable: ", d.definedConst);
+  writeln("Using instance: ", d._value.definedConst);
+
+  writeln("Array on domain: ", arr);
+}
+
+writeln("var domain inited with literal");
+var varDom = {1..10};
```

Comment:
It seems we could have the compiler look at uses of `varDom` in cases like this and adjust it to be `const` if nothing relied on it not being `const`. This would be good to describe in future work.